code/espurna/garland.cpp

```diff
@@ -178,59 +252,209 @@ void _garlandWebSocketOnAction(uint32_t client_id, const char* action, JsonObjec
+    _currentAnim     = new_anim;
+    _currentPalette  = new_palette;
+    palette_name = _currentPalette->name();
+    DEBUG_MSG_P(PSTR("[GARLAND] Anim: %-10s Pal: %-8s Inter: %d\n"),
+                _currentAnim->name(), palette_name.c_str(), _currentDuration);
+
+    scene.setAnim(_currentAnim);
+    scene.setPalette(_currentPalette);
+    scene.setup();
+}
+
+//------------------------------------------------------------------------------
+bool executeCommand(const String& command) {
+    DEBUG_MSG_P(PSTR("[GARLAND] Executing command \"%s\"\n"), command.c_str());
+    // Parse JSON input
+    DynamicJsonBuffer jsonBuffer;
+    JsonObject& root = jsonBuffer.parseObject(command);
+    if (!root.success()) {
+        DEBUG_MSG_P(PSTR("[GARLAND] Error parsing command\n"));
+        return false;
+    }
+
+    bool scene_setup_required = false;
+
+    if (root.containsKey(MQTT_PAYLOAD_ENABLE)) {
+        auto enable = root[MQTT_PAYLOAD_ENABLE].as<String>();
+        garlandEnabled(enable != "false");
+    }
+
+    if (root.containsKey(MQTT_PAYLOAD_BRIGHTNESS)) {
+        auto brightness = root[MQTT_PAYLOAD_BRIGHTNESS].as<byte>();
+        scene.setBrightness(brightness);
+    }
+
+    if (root.containsKey(MQTT_PAYLOAD_ANIM_SPEED)) {
+        auto speed = root[MQTT_PAYLOAD_ANIM_SPEED].as<byte>();
+        scene.setSpeed(speed);
+    }
+
+    Anim* newAnim = _currentAnim;
+    if (root.containsKey(MQTT_PAYLOAD_ANIMATION)) {
+        auto animation = root[MQTT_PAYLOAD_ANIMATION].as<const char*>();
+        for (size_t i = 0; i < animsSize(); ++i) {
+            auto anim_name = anims[i]->name();
+            if (strcmp(animation, anim_name) == 0) {
+                newAnim = anims[i];
+                scene_setup_required = true;
+                break;
+            }
+        }
+    }
+
+    Palette* newPalette = _currentPalette;
+    if (root.containsKey(MQTT_PAYLOAD_PALETTE)) {
+        if (root.is<int>(MQTT_PAYLOAD_PALETTE)) {
+            one_color_palette.reset(new Palette("Color", {root[MQTT_PAYLOAD_PALETTE].as<uint32_t>()}));
+            newPalette = one_color_palette.get();
+        } else {
+            auto palette = root[MQTT_PAYLOAD_PALETTE].as<const char*>();
+            bool palette_found = false;
+            for (size_t i = 0; i < palsSize(); ++i) {
+                auto pal_name = pals[i].name();
+                if (strcmp(palette, pal_name) == 0) {
+                    newPalette = &pals[i];
+                    palette_found = true;
+                    scene_setup_required = true;
+                    break;
+                }
+            }
+            if (!palette_found) {
+                uint32_t color = (uint32_t)strtoul(palette, NULL, 0);
+                if (color != 0) {
+                    one_color_palette.reset(new Palette("Color", {color}));
+                    newPalette = one_color_palette.get();
+                }
+            }
+        }
+    }
+
+    unsigned long newAnimDuration = LONG_MAX;
```

Comment:
ULONG_MAX?
also, see https://en.cppreference.com/w/cpp/types/numeric_limits

e.g. `std::numeric_limits<unsigned long>::max()` (or, when needed, `decltype(newAnimDuration)` to avoid issues)
`unsigned long var { ... };` init will also warn about the mistypes like that, just wary of the `auto` that will be deduced into a list instead of the var, b/c we use such an old gcc version